sros2/test/sros2/commands/security/verbs/utilities/sros2_cli_test_case.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import contextlib
+import unittest
+
+from launch import LaunchDescription
+from launch.actions import ExecuteProcess
+from launch.actions import TimerAction
+
+import launch_testing
+import launch_testing.asserts
+import launch_testing.markers
+import launch_testing.tools
+import launch_testing_ros.tools
+
+
+MAX_DISCOVERY_DELAY = 2.0  # seconds
+
+
+def generate_sros2_cli_test_description(
+    fixture_actions, rmw_implementation, use_daemon
+):
+    additional_env = {'RMW_IMPLEMENTATION': rmw_implementation}
+    if use_daemon:
+        # Start daemon.
+        fixture_actions = [ExecuteProcess(
+            cmd=['ros2', 'daemon', 'start'],
+            name='daemon-start',
+            # Wait for daemon to discover fixture nodes.
+            on_exit=[TimerAction(
+                period=MAX_DISCOVERY_DELAY,
+                actions=fixture_actions
+            )],
+            additional_env=additional_env
+        )]
+    return LaunchDescription([
+        # Always stop daemon to avoid cross-talk.
+        ExecuteProcess(
+            cmd=['ros2', 'daemon', 'stop'],
+            name='daemon-stop',
+            on_exit=fixture_actions,
+            additional_env=additional_env
+        ),
+    ])
+
+
+class SROS2CLITestCase(unittest.TestCase):
```

Comment:
The existing tests in this project are all pytest. Why are we using unittest here?

Reply:
Because `test_generate_policy*.py` is no longer a `pytest`, but a `pytest`-run `launch` test. And `launch_testing` is built on top of `unittest`. There's a desire to achieve a stronger integration between `launch_testing` and `pytest` (even turning the former into a plugin for the latter, see https://github.com/ros2/launch/issues/405), but as of today, we can only include `launch` tests in `pytest` runs.

Reply:
Alright, thank you for explaining. I really dislike mixing the two, here. It's complicated enough having both of them in existence and sharing space in my head-- having them both used in the same project will cause no end of confusion. Is there any other way to do this?

Reply:
There's not, assuming that testing against multiple RMW implementations is a requirement (I understand it is based on https://github.com/ros2/sros2/pull/212#issuecomment-628245056) and that we're not looking forward to recreate `launch` and `launch_testing` using `subprocess` and `pytest` just for this one test. I'd rather not do that.